File: seed/php-sdk/server-sent-events-openapi/src/Types/CompletionStreamChunk.php
<?php

namespace Seed\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

/**
 * A single chunk in a streamed completion response.
 */
class CompletionStreamChunk extends JsonSerializableType
{
    /**
     * @var ?string $delta The incremental text chunk.
     */
    #[JsonProperty('delta')]
    public ?string $delta;

    /**
     * @var ?int $tokens Number of tokens in this chunk.
     */
    #[JsonProperty('tokens')]
    public ?int $tokens;

    /**
     * @param array{
     *   delta?: ?string,
     *   tokens?: ?int,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->delta = $values['delta'] ?? null;
        $this->tokens = $values['tokens'] ?? null;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: seed/csharp-sdk/server-sent-events-openapi/src/SeedApi/Types/CompletionStreamChunk.cs
using global::System.Text.Json;
using global::System.Text.Json.Serialization;
using SeedApi.Core;

namespace SeedApi;

/// <summary>
/// A single chunk in a streamed completion response.
/// </summary>
[Serializable]
public record CompletionStreamChunk : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    /// <summary>
    /// The incremental text chunk.
    /// </summary>
    [JsonPropertyName("delta")]
    public string? Delta { get; set; }

    /// <summary>
    /// Number of tokens in this chunk.
    /// </summary>
    [JsonPropertyName("tokens")]
    public int? Tokens { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
